ext/serializer.c

```diff
@@ -1808,6 +1788,28 @@ void ddtrace_serialize_span_to_array(ddtrace_span_data *span, zval *array) {
         }
     }
 
+    if (inferred_span) {
+        ddtrace_serialize_span_to_array(inferred_span, array);
```

Comment:
You could make `ddtrace_serialize_span_to_array` return the inserted array instead of refetching from `Z_ARR_P(array)`.